sdk/eventhub/Azure.Messaging.EventHubs/src/Core/TransportProducerPool.cs

```diff
@@ -101,17 +101,17 @@ internal TransportProducerPool()
                 return new PooledProducer(EventHubProducer);
             }
 
-            var identifier = Guid.NewGuid().ToString();
-            var item = Pool.GetOrAdd(partitionId, id => new PoolItem(partitionId, TransportProducerFactory(id), removeAfterDuration));
+            var identifier = Guid.NewGuid();
+            var item = Pool.GetOrAdd(partitionId, id => new PoolItem(id, TransportProducerFactory(id), removeAfterDuration));
```

Comment:
It would be nice to be able to properly #ifdef here to that the `GetOrAdd` method could be used that allows to pass the argument. Then the closure allocation would be gone because the pool could pass itself as an argument to the value factory. It would require to slightly change the remove after duration value propagation but since the code currently always calls ExtendRemoveAfter it would be easy to rename that method and bring in the constructor logic that checks for the default value.

Reply:
not sure that I'm following this one...

Reply:
I'll try to type some up pseudo code later.

Reply:
hrmm...   this was a contribution, so I don't know all the ins-and-outs as well as I'd like, but I don't see a reason why we couldn't remove the duration for the initial construction.    We'd end up with a default value (10 minutes) and, to your point, we do explicitly extend it again once we get an instance.  

We'd probably want to tweak the default duration to 5 minutes to match what the producer uses.   Not sure if that throws off the tests at all, but I'm hoping not.

Reply:
These changes would be good if we manage to bring in the GetOrAdd overload for TFMs that support it. Are other parts of the SDK doing more TFM specific optimizations? Is that something OK doing?

Reply:
No you are absolutely right and I agree at this stage it is probably not worth the hazzle.